akka-http/src/main/scala/akka/http/scaladsl/server/HttpApp.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.server
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.stream.ActorMaterializer
+import akka.http.scaladsl.settings.ServerSettings
+import com.typesafe.config.{ Config, ConfigFactory }
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{ Await, ExecutionContext, Future }
+import scala.io.StdIn
+import scala.util.{ Failure, Success }
+
+/**
+ * Bootstrap trait. It helps booting up an akka-http by only defining the desired routes.
+ * It offers additional hooks to modify the default behavior
+ */
+trait HttpApp {
+
+  def main(args: Array[String]) = {
+    val config = createConfig()
+    implicit val system = createActorSystem(config)
+    implicit val materializer = ActorMaterializer()
+    implicit val executionContext = system.dispatcher
+    postStartActorSystem(system)
+
+    val host = config.getString("akka.http.server.host")
+    val port = config.getInt("akka.http.server.port")
+
+    val bindingFuture = Http().bindAndHandle(
+      handler = route,
+      interface = host,
+      port = port,
+      settings = ServerSettings(config))
+
+    bindingFuture.onComplete {
+      case Success(_) ⇒
+        println(s"Server online at http://$host:$port/")
+      case Failure(cause) ⇒
+        println(s"Error starting the server $cause")
+        preTerminateActorSystem(system)
+        system.terminate()
+    }
+
+    //It's waiting forever because maybe there is never a shutdown signal
+    Await.result(waitForShutdownSignal(system), Duration.Inf)
+
+    bindingFuture
+      .flatMap(_.unbind())
+      .onComplete(_ ⇒ {
+        preTerminateActorSystem(system)
+        system.terminate()
+      })
+  }
+
+  /**
+   * Provides the [[ActorSystem]] from where the routes will be served. Override this method, if the default [[ActorSystem]] doesn't suit your needs. You don't need to store the created instance of the [[ActorSystem]].
+   */
+  protected def createActorSystem(config: Config): ActorSystem = ActorSystem()
```

Comment:
shouldn't this be passing in the config?
Give the Actor System a name - via `Logging.simpleName(getClass)`?